activerecord/lib/active_record/railties/databases.rake

```diff
@@ -84,10 +84,60 @@ db_namespace = namespace :db do
     ActiveRecord::Tasks::DatabaseTasks.purge_current
   end
 
+  # Do I want a task here or is having a bin/safety_check runner sufficient? I think a runner is okay for now.
+  # I just want a method in the migration flow that outputs the resulting SQL
+  #     Maybe a thing that works ENV['DRY_RUN_MIGRATION'] == true that does the same?
+  # task :before_migrate do
+  #   Rails.logger.debug "before migrate is working"
+  #   ActiveRecord::Tasks::DatabaseTasks.pre_migrate
+  # end
+
+
+  # This is kinda weird since we should allow read queries to go through so that Active Record models can be used in migrations but we don’t want to allow writes.
+  #
+  # This almost works – I also want the transaction to fail / the ActiveRecord::SchemaMigration Create to not happen since nothing should be migrated
+  #   Where does this happen? It happens in Migration#record_version_state_after_migrating
+  #
+  # However – if the migration is also creating the schema_migrations table, that’s bad
+  #
+  # Should I use a refinement instead?
+  #
+  # Where is the spot that I want to allow the schema_migrations table being created?
+  #
+  module DryConnection
+    def execute(sql, name = nil)
+      # So that I can turn this off after the pre-migration task
+      if ENV["DRY_RUN_MIGRATION"]
```

Comment:
This check is here to support the use case of wanting to have a dry-run in the pre-migration task for something like running safety checks / validations on the results of the migration, and then disable the dry-run after.

Is this a good use case for refinements to handle the on/off behaviour?